Makefile

```diff
@@ -1,18 +1,34 @@
-.PHONY: dev clean e2e push
+.PHONY: clean dev-public-cluster update-image-on-deployment push-tester-image deploy-e2e run-e2e
 
-dev:
-	terraform --version
-	cd devenv && mkdir -p state && terraform init && terraform apply -auto-approve
 
-clean:
-	rm -rf devenv/state devenv/.terraform.lock.hcl devenv/.terraform devenv/terraform.tfstate devenv/terraform.tfstate.backup
+# keep separate for simultaneous public/private dev without need for resource recreation
+clean-public:
+	rm -rf devenv/state devenv/public_cluster_tf/.terraform.lock.hcl devenv/public_cluster_tf/.terraform devenv/public_cluster_tf/terraform.tfstate devenv/public_cluster_tf/terraform.tfstate.backup
 
-e2e:
-	go test -v --count=1 --tags=e2e ./e2e
+dev-public-cluster:
+	terraform --version
+	cd devenv && mkdir -p state && cd public_cluster_tf && terraform init && terraform apply -auto-approve
 
-push:
+# aka make push (formerly known as make push-private)
+update-image-on-deployment:
 	echo "$(shell cat devenv/state/registry.txt)/app-routing-operator:$(shell date +%s)" > devenv/state/operator-image-tag.txt
 	az acr login -n `cat devenv/state/registry.txt`
 	docker build -t `cat devenv/state/operator-image-tag.txt` .
 	docker push `cat devenv/state/operator-image-tag.txt`
-	kubectl set image -n kube-system --kubeconfig devenv/state/kubeconfig deployments/app-routing-operator operator=`cat devenv/state/operator-image-tag.txt`
+	cd devenv && /usr/bin/env sh scripts/push_image.sh
+
+push-tester-image:
+	# grab the image name and tag - that first gets determined by where and how we push it - so first need to build and push it - then export that name and registry and tag into the YAML
+	az acr login -n `cat devenv/state/registry.txt`
+	echo "$(shell cat devenv/state/registry.txt)/app-routing-operator-e2e:$(shell date +%s)" > devenv/state/e2e-image-tag.txt
+	docker build --platform=linux/amd64 -t `cat devenv/state/e2e-image-tag.txt` -f e2e/Dockerfile .
+	docker push `cat devenv/state/e2e-image-tag.txt`
+	export IMAGE=`cat devenv/state/e2e-image-tag.txt` && cd devenv && envsubst < e2e-tester.yaml > state/e2e-tester-formatted.yaml
```

Comment:
Using kustomize instead of this would make sense. That's what that tool is for and it's more flexible + native than bash

Reply:
I feel like leaving it in bash is easier to read/understand what's happening without needing an external tool, plus I feel like kustomize is really useful when there's shared state across lots of different files but since we're only doing this for two files I'm not sure that the overhead of setting up kustomize for this is worth it.

Also, if I'm not mistaken, we'd just need to write the same values to kustomization files that would hold the values, and we'd need the same kind of bash for that too right?

I'm not super familiar with kustomize though, reading up on it now. Can look more into it if you feel strongly about this!

Reply:
I'm okay with leaving this for now but I think Oliver makes a good point here, we can probably simplify this in the future!

Reply:
makes sense - I totally see the appeal and have something partially working, will try spending a bit longer on it and see how far I can get

Reply:
Yeah this is fine for now but something good to keep in mind for the future

Reply:
update: using kustomize now, so marking this as resolved